### DISCUSSION TITLE
M2M Not generated in the Client

### DISCUSSION THREAD

**Discussion by cupcakearmy** [prisma/prisma#7911] [Q&A] [answered]
Hi everyone,

I am using Prisma for the first time and I'm really liking it so far. 
There a small issue, I'm probably missing something. I want to have an [implicit m2m relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#implicit-many-to-many-relations). However if i run migrations and regenerate the client it does not show up in the client.

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

enum UserRole {
  USER
  ADMIN
}

model User {
  id       String   @id @default(uuid())
  email    String   @unique
  password String
  role     UserRole @default(USER)

  hubs Hub[]
}

model Hub {
  id     String  @id @default(uuid())
  name   String  @unique
  events Event[]

  members User[]
}

// ...
```

Any ideas why?
Thanks!

**Comment by ryands17** [comment, marked as answer]
@cupcakearmy 👋 

I tried this and it works fine when I run the migrations. Could you provide a sample repo so that I can check?

**Reply by cupcakearmy** [reply]
By creating a repro repo i noticed that it actually works and types are included if I include the relation in the `include` object.
It's good design on your part, perfect typings. Maybe the only thing I would add is some kind of example in the [docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#implicit-many-to-many-relations) For someone new to prisma the `include` object might not be familiar.
Thanks :)

```ts
import { PrismaClient } from '@prisma/client'

async function main() {
  const client = new PrismaClient()
  const user = await client.user.findFirst({ include: { hubs: true } })
  console.log(user)
}

main()
```

**Reply by ryands17** [reply]
We have those in the docs [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries). You can get here by searching for the keyword **relation**.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
